dvc/output/base.py

```diff
@@ -195,17 +196,35 @@ def get_hash(self):
                 self.path_info,
                 self.fs,
                 self.fs.PARAM_CHECKSUM,
+                dvcignore=self.dvcignore,
             ).hash_info
         return ostage(
-            self.odb, self.path_info, self.fs, self.odb.fs.PARAM_CHECKSUM
+            self.odb,
+            self.path_info,
+            self.fs,
+            self.odb.fs.PARAM_CHECKSUM,
+            dvcignore=self.dvcignore,
         ).hash_info
 
     @property
     def is_dir_checksum(self):
         return self.hash_info.isdir
 
+    def _check_path_dvcignore(self, path) -> bool:
+        if (
+            self.scheme == Schemes.LOCAL
+            and self.IS_DEPENDENCY is False
+            and self.repo
+        ):
+            if self.repo.dvcignore.is_ignored(path, ignore_subrepos=False):
```

Comment:
We should probably pass `fs` to `is_ignored` too, so it could decide to just shortcircuit if it is not a local filesystem.

Reply:
Also, should we use `self.dvcignore` here instead of `self.repo.dvcignore`?